### PR TITLE
P1638R1 basic_istream_view::iterator should not be copyable

### LINKED ISSUE #3028: [2019-07 LWG Motion 24] P1638R1 basic_istream_view::iterator should not be copyable

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3028.

I wanted this pull request to just contain the patch against P1035R7, but that apparently failed.

### CODE REVIEW

**Comment by johelegp** [comment]
That's because you want to merge into `master`, which doesn't have P1035R7.

File: source/ranges.tex [hunk 1]
```diff
@@ -3802,1398 +3855,2476 @@
...
+\begin{itemdecl}
+constexpr explicit inner_iterator(outer_iterator<Const> i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Initializes \tcode{i_} with \tcode{i}.
+\end{itemdescr}
+
+\indexlibr
```
Comment (johelegp) [inline source/ranges.tex]: Remove space between `iterator` and `&`.
```suggestion
    iterator& operator=(const iterator&) = delete;
```

Comment (johelegp) [inline source/ranges.tex]: Same as above.
```suggestion
    iterator& operator=(iterator&&) = default;
```

Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

Comment (CaseyCarter) [inline source/ranges.tex]:
```suggestion
    iterator& operator=(const iterator&) = delete;
```

Comment (CaseyCarter) [inline source/ranges.tex]:
```suggestion
    iterator& operator=(iterator&&) = default;
```

Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

**Comment by burblebee** [comment]
@CaseyCarter can you approve this?

**Comment by burblebee** [comment]
> The changes this PR applies from P1638R1 - the last three commits - are correct, but the PR also includes P1035 changes and **should not be merged without rebasing onto the result of merging #3106.**
> 
> @burblebee, Could we label this rebase-required to ensure no one merges it without performing that rebase?

@CaseyCarter, @zygoloid will do this when he merges this motion.

**Comment by johelegp** [comment]
Or do as @zygoloid did with Text Formatting, and add it to the same PR.

**Pull request merged**